spec/rails_integration_spec.rb

```diff
@@ -106,14 +106,13 @@ def prepare_params(params)
   end
 
   describe "nested_array" do
-    it "responds with a 400 when the nested array is not supplied properly" do
-      params = {
-        'filter' => 'state'
-      }
+    it "responds with a 200 when the nested array is provided as nil" do
+      expect { get :nested_array, **prepare_params({ filter: { state: nil } }) }.not_to raise_error
```

Comment:
this will pass the following to recurrent calls to `ParamEvaluator`:

```
#<ActionController::Parameters {"filter"=>{"state"=>""}, "controller"=>"fake", "action"=>"nested_array"} permitted: false>
#<ActionController::Parameters {"state"=>""} permitted: false>
```

Reply:
This looks like Rails getting in the way. Since we're calling a GET endpoint, the parameters are encoded in a query-string format and when decoded the value of state is `''` instead of nil.
We should either test this with a `post` endpoint, or mock `controller.params` like we do in other tests:
```ruby
allow(controller).to receive(:params).and_return({ filter: { state: nil } })
```

Reply:
@iMacTia I have done a test where I changed the `nested_array` to a `post` endpoint and the result is the same:

```
#<ActionController::Parameters {"filter"=>{"state"=>""}, "controller"=>"fake", "action"=>"nested_array"} permitted: false>
#<ActionController::Parameters {"state"=>""} permitted: false>
```

Reply:
This is super strange, where is the empty string coming from?!